pynest/nest/lib/hl_api_types.py

```diff
@@ -880,3 +883,51 @@ def apply(self, spatial_nc, positions=None):
                 if len(pos) != len(positions[0]):
                     raise ValueError('All positions must have the same number of dimensions')
             return sli_func('Apply', self._datum, {'source': spatial_nc, 'targets': positions})
+
+
+def serializable(data):
+    """Make data serializable for JSON.
+
+    Parameters
+    ----------
+    data : any
+
+    Returns
+    -------
+    data_serialized : str, int, float, list, dict
+        Data can be encoded to JSON
+    """
+
+    if isinstance(data, (numpy.ndarray, NodeCollection)):
+        data_serialized = data.tolist()
+    elif isinstance(data, SynapseCollection):
+        # Get full information from SynapseCollection
+        data_serialized = serializable(data.get())
+    elif isinstance(data, kernel.SLILiteral):
+        # Get name of SLILiteral.
+        data_serialized = data.name
+    elif isinstance(data, (list, tuple)):
+        data_serialized = [serializable(d) for d in data]
+    elif isinstance(data, dict):
+        data_serialized = dict([(key, serializable(value)) for key, value in data.items()])
+    else:
```

Comment:
This looks fragile: return type might not be str, int or float if I pass an instance of a non-serializable class. It's not immediately obvious to me how to improve it; e.g. to check for "floatness" would also require testing against numpy data types. Perhaps we could attempt to typecast to str or float inside a try..except block.

Reply:
Which non-serializable objects do you have in mind? And how shall it make those serializable for json? 

Even non-number (bool, str) numpy arrayis working for json when array is converted to list.